### PR TITLE
Issue #570 Print a reason why root access is needed before using it

### LINKED ISSUE #570: Print a message before asking for sudo password
People have raised concerns as to why `crc` asks for root access during setup, this issue is to have `crc` print a reason before asking/using root access using `sudo`.

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #570

### CODE REVIEW

**Comment by praveenkumar** [comment]
@anjannath you also need to fix the integration test along with those changes.

File: pkg/os/exec.go [hunk 1]
```diff
@@ -16,6 +19,7 @@ func RunWithPrivilege(cmdAndArgs ...string) (string, string, error) {
 	stdErr := new(bytes.Buffer)
 	cmd.Stdout = stdOut
 	cmd.Stderr = stdErr
+	logging.Infof("Will use root access to %s\n", reason)
```
Comment (cfergeau) [inline pkg/os/exec.go]: I'd prefer that we avoid creating full sentences like this, this is going to be error prone if one does not pay attention how this works. "Will use root access: %s" would work better I think.

File: test/integration/features/basic.feature [hunk 2]
```diff
@@ -29,17 +29,32 @@ Feature: Basic test
     @linux
```
[the remark this answers is not in the document]
Comment (anjannath) [reply test/integration/features/basic.feature]: Actually it doesn't break the integration tests, but right, i'll add them to the earlier commit.

File: pkg/crc/preflight/preflight_linux.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func SetupHost(vmDriver string) {
 	preflightCheckAndFix(false,
 		checkIfRunningAsNormalUser,
 		fixRunAsNormalUser,
-		"Checking if running as root",
+		"Checking if running as non-root",
```
[the remark this answers is not in the document]
Comment (anjannath) [reply pkg/crc/preflight/preflight_linux.go]: I guess he meant to change `non-root` => `normal user` like in windows, am ok with either, which sounds better ;)

[the remark this answers is not in the document]
Comment (cfergeau) [reply pkg/crc/preflight/preflight_linux.go]: yes that's what I meant, just a suggestion, just pick the wording you like best :)

**Pull request merged**